### DISCUSSION TITLE
Using node modules server side

### DISCUSSION THREAD

**Discussion by etfz** [nuxt/nuxt#15949] [Questions]
Hi,

Familiar with Vue and Javascript/node, new to Nuxt. I'm having great difficulty understanding the fundamentals of how scripting works. I want to use some node modules on a page basis (server side), but I have no idea how to do so. Am I not meant to import them in the page's setup script? Trying to import [whois-json](https://www.npmjs.com/package/whois-json) like so:
```
<script setup>

import whois from 'whois-json'

</script>

<template>
	
</template>
```
gives me the error `error loading dynamically imported module`. Same with a few other NPM modules I tried. Importing builtin node modules seems to work. Trying to import some simple node (ES) modules I made myself yields `global is not defined`.

I feel like I've missed something obvious.

As a bonus (unless it becomes apparent when I've learned how the above works) I am also interested in how to import "local" files as modules. Again server side. For example just a simple module that exports a single class. All I can find is documentation on Nuxt modules and plugins, which I don't *think* is what I want.

**Comment by etfz** [comment]
Shameless bump and TLDR: How do I use modules installed using NPM? I get either `error loading dynamically imported module` or `global is not defined`, depending on module. The modules are made for node.js.

**Reply by Baroshem** [reply]
Hey, I think I might have an answer for you. Could you please create a reproduction on Stackblitz?

It will be easier to show on the repro example :)

**Reply by etfz** [reply]
Hi,

Well, it seems a bit inconsistent now when I try to reproduce it in new projects, but I made this.
https://stackblitz.com/edit/nuxt-starter-awfmhg?file=pages%2Ftest.vue,package.json
This is a new error now, but at least I get the same in a new project I made locally. (at `/test`)
```
class heritage events_1.EventEmitter is not an object or null
```
Tried installing the `events` package as per someone's suggestion, which upgraded the error to the following:
```
assignment to undeclared variable log
```

**Comment by Baroshem** [comment]
@etfz 

In the place where you are importing the `whois-json` package, if you replace the classic importing with the following, the error should be gone, and you should be able to use this package the way it is meant to be :)

```ts
const whois = await import('whois-json').then((lib) => lib.default || lib);
// instead of "import whois from 'whois-json'"
```

Below, the link to stackblitz repro -> https://stackblitz.com/edit/nuxt-starter-a2r44g?file=pages%2Ftest.vue,components%2FTest1.vue

**Reply by etfz** [reply]
Thank you! I'll try this. Is there any documentation regarding this phenomenon, or why does this happen and when do I need to resort to this solution?

**Reply by Baroshem** [reply]
This docs section could help you :) 

https://nuxt.com/docs/guide/concepts/esm#default-exports
